tests/dialects/test_snowflake.py

```diff
@@ -519,6 +519,12 @@ def test_snowflake(self):
                 "snowflake": "ALTER TABLE foo ADD COLUMN id INT AUTOINCREMENT START 1 INCREMENT 1",
             },
         )
+        self.validate_all(
+            "ALTER TABLE foo ADD CONSTRAINT c1 UNIQUE (email)",
+            write={
+                "snowflake": "ALTER TABLE foo ADD CONSTRAINT c1 UNIQUE (email)",
+            },
+        )
```

Comment:
We usually write tests like this, where the read and write dialects are the same, using `validate_identity`:

```python
self.validate_identity("ALTER TABLE foo ADD CONSTRAINT c1 UNIQUE (email)")
```

Let's make this change and move it appropriately so it's grouped with the rest of the `identity` checks (nit).